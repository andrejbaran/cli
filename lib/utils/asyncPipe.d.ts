/**
 * @author: JP Lew (jp@cto.ai)
 * @date: Tuesday, 23rd April 2019 11:05:01 pm
 * @lastModifiedBy: JP Lew (jp@cto.ai)
 * @lastModifiedTime: Wednesday, 24th April 2019 12:45:12 pm
 * @copyright (c) 2019 CTO.ai
 */
declare const asyncPipe: (...fns: Function[]) => (param: any) => any;
declare const _trace: (msg: string) => (x: any) => any;
export { asyncPipe, _trace };
