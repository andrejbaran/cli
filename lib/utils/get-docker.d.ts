import Docker from 'dockerode';
/**
 * Helper function to check the existence of docker
 * @param self Is the instance of 'this'
 * @param type Is the context of the caller, so we know which error to log
 */
export default function getDocker(self: any, type: string): Promise<Docker | undefined>;
