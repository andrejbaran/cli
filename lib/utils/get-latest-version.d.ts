/**
 * Author: Brett Campbell (brett@hackcapital.com)
 * Date: Friday, 12th April 2019 10:33:26 am
 * Last Modified By: Brett Campbell (brett@hackcapital.com)
 * Last Modified Time: Friday, 12th April 2019 10:33:26 am
 *
 * DESCRIPTION
 *
 */
export default function getLatestVersion(): Promise<string | undefined>;
