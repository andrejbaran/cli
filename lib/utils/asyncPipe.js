"use strict";
/**
 * @author: JP Lew (jp@cto.ai)
 * @date: Tuesday, 23rd April 2019 11:05:01 pm
 * @lastModifiedBy: JP Lew (jp@cto.ai)
 * @lastModifiedTime: Wednesday, 24th April 2019 12:45:12 pm
 * @copyright (c) 2019 CTO.ai
 */
Object.defineProperty(exports, "__esModule", { value: true });
// based on https://github.com/obengwilliam/pipeawait
const asyncPipe = (...fns) => (param) => fns.reduce(async (acc, fn) => fn(await acc), param);
exports.asyncPipe = asyncPipe;
const _trace = (msg) => (x) => {
    console.log(msg, x);
    return x;
};
exports._trace = _trace;
