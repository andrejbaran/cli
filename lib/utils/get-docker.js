"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const sdk_1 = require("@cto.ai/sdk");
const dockerode_1 = tslib_1.__importDefault(require("dockerode"));
const fs = tslib_1.__importStar(require("fs-extra"));
const env_1 = require("../constants/env");
/**
 * Helper function to display appropriate error message to the user
 * @param self Is the instance of 'this'
 * @param numRepeats Checks the number of retries the user has attempted
 * @param type The type of error message we want to display
 */
function logError(self, numRepeats, type) {
    switch (type) {
        case 'account-create-docker-not-installed': {
            if (!numRepeats) {
                self.log(`\n${sdk_1.ux.colors.reset.cyan("We're almost there! You'll just need to install Docker for CTO.ai ops to run properly - go here to install it now.")}`);
                self.log(`\n${sdk_1.ux.colors.reset.green('→')} https://docs.docker.com/install/`);
                self.log(`${sdk_1.ux.colors.reset.grey("(You'll create an account with Docker in order to start the download)")}`);
                self.log(`\nOnce installed, make sure you start the Docker app, then come back to this terminal and type ${sdk_1.ux.colors.reset.cyan("'Y'")}.`);
                self.log("We'll be waiting right here when you're ready 👍\n");
            }
            else if (numRepeats >= 3) {
                self.log(`\n${sdk_1.ux.colors.reset.cyan("Hmm. Docker still doesn't seem to be running.")}`);
                self.log(`${sdk_1.ux.colors.reset.cyan('Please check again, or run, "ops account support" and we\'ll be happy to help you out.')}`);
            }
            else {
                self.log(`\n${sdk_1.ux.colors.reset.cyan('Please check that Docker is running again and come back here.')}`);
            }
            break;
        }
        case 'account-create-docker-stopped': {
            if (!numRepeats) {
                self.log(`\n${sdk_1.ux.colors.reset.cyan("It looks like you have Docker installed, but it's not currently running.")}`);
                self.log(`${sdk_1.ux.colors.reset.cyan('Please start the Docker app to continue (You can find it in the MacOS → Applications folder)')}`);
                self.log(`\nOnce Docker is running, come back to this terminal and type ${sdk_1.ux.colors.reset.cyan("'Y'")}`);
                self.log("We'll be waiting right here when you're ready 👍\n");
            }
            else if (numRepeats >= 3) {
                self.log(`\n${sdk_1.ux.colors.reset.cyan("Hmm. Docker still doesn't seem to be running.")}`);
                self.log(`${sdk_1.ux.colors.reset.cyan('Please check again, or run, "ops account support" and we\'ll be happy to help you out.')}`);
            }
            else {
                self.log(`\n${sdk_1.ux.colors.reset.cyan('Please check that Docker is running again and come back here.')}`);
            }
            break;
        }
        default:
            if (type.endsWith('docker-not-installed')) {
                if (!numRepeats) {
                    self.log(`\n${sdk_1.ux.colors.reset.cyan("Uh-oh! You'll need to install and run Docker for CTO.ai ops to work properly - go here to install it now.")}`);
                    self.log(`\n${sdk_1.ux.colors.reset.green('→')} https://docs.docker.com/install/`);
                    self.log(`${sdk_1.ux.colors.reset.grey("(You'll create an account with Docker in order to start the download)")}`);
                    self.log(`\nOnce installed, make sure you start the Docker app, then come back to this terminal and type ${sdk_1.ux.colors.reset.cyan("'Y'")}.`);
                    self.log("We'll be waiting right here when you're ready 👍\n");
                }
                else if (numRepeats >= 3) {
                    self.log(`\n${sdk_1.ux.colors.reset.cyan("Hmm. Docker still doesn't seem to be running.")}`);
                    self.log(`${sdk_1.ux.colors.reset.cyan('Please check again, or run, "ops account support" and we\'ll be happy to help you out.')}`);
                }
                else {
                    self.log(`\n${sdk_1.ux.colors.reset.cyan('Please check that Docker is running again and come back here.')}`);
                }
            }
            else {
                if (!numRepeats) {
                    self.log(`\n${sdk_1.ux.colors.reset.cyan("It looks like you have Docker installed, but it's not currently running.")}`);
                    self.log(`${sdk_1.ux.colors.reset.cyan('Please start the Docker app to continue (You can find it in the MacOS → Applications folder)')}`);
                    self.log(`\nOnce Docker is running, come back to this terminal and type ${sdk_1.ux.colors.reset.cyan("'Y'")}`);
                    self.log("We'll be waiting right here when you're ready 👍\n");
                }
                else if (numRepeats >= 3) {
                    self.log(`\n${sdk_1.ux.colors.reset.cyan("Hmm. Docker still doesn't seem to be running.")}`);
                    self.log(`${sdk_1.ux.colors.reset.cyan('Please check again, or run, "ops account support" and we\'ll be happy to help you out.')}`);
                }
                else {
                    self.log(`\n${sdk_1.ux.colors.reset.cyan('Please check that Docker is running again and come back here.')}`);
                }
            }
    }
}
/**
 * Helper function to display the prompt to the user if they want to retry
 */
async function confirmReadyContinue() {
    return sdk_1.ux.prompt({
        type: 'confirm',
        name: 'answer',
        message: `${sdk_1.ux.colors.reset.cyan('Ready to continue?')}`,
    });
}
/**
 * Helper function to check the existence of docker
 * @param self Is the instance of 'this'
 * @param type Is the context of the caller, so we know which error to log
 */
async function getDocker(self, type) {
    // Point to the docker socket
    /**
     * Checks whether docker is installed on the machine
     * Keep repeating until the socket is available
     */
    let numRepeats = 0; // Flag to set whether the prompt has been repeated or not
    let isDockerInstalled = false; // Flag to indicate whether docker is installed or not
    while (!isDockerInstalled) {
        try {
            const stats = fs.statSync(env_1.DOCKER_SOCKET); // Resets the status of the socket
            // Check the validity of the socket, which is only available if docker is installed
            isDockerInstalled = stats.isSocket();
        }
        catch (_a) {
            logError(self, numRepeats, `${type}-docker-not-installed`); // Log the error message
            // Ask the user for prompt
            let { answer } = await confirmReadyContinue();
            // Return if user doesn't want to repeat
            if (!answer)
                process.exit(0); // Exit safely if there is no docker without an error
            numRepeats++; // Adds the number of errors the user has repeated
        }
    }
    /**
     * Checks whether docker is running
     * At this point, assume docker has already been installed
     */
    numRepeats = 0; // Re-initialize the flag for repeat
    let isDockerRunning = false; // Flag to indicate whether docker is running or not
    let docker; // Initialize return variable
    while (!isDockerRunning) {
        // Instantiate a new docker client
        docker = new dockerode_1.default({ socketPath: env_1.DOCKER_SOCKET });
        try {
            await docker.ping(); // Try to ping the docker daemon
            // If successful, set flag to true to exit the while loop
            isDockerRunning = true;
        }
        catch (_b) {
            logError(self, numRepeats, `${type}-docker-stopped`);
            // Ask the user for prompt
            let { answer } = await confirmReadyContinue();
            // Return if user doesn't want to repeat
            if (!answer)
                process.exit(0); // Exit safely if there is no docker without an error
            numRepeats++; // Adds the number of errors the user has repeated
        }
    }
    return docker;
}
exports.default = getDocker;
