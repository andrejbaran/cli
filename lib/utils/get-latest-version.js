"use strict";
/**
 * Author: Brett Campbell (brett@hackcapital.com)
 * Date: Friday, 12th April 2019 10:33:26 am
 * Last Modified By: Brett Campbell (brett@hackcapital.com)
 * Last Modified Time: Friday, 12th April 2019 10:33:26 am
 *
 * DESCRIPTION
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const axios_1 = tslib_1.__importDefault(require("axios"));
const customErrors_1 = require("../errors/customErrors");
async function getLatestVersion() {
    try {
        const { data } = await axios_1.default({
            method: 'GET',
            url: '/@cto.ai/ops',
            baseURL: 'https://registry.npmjs.org/',
        });
        const { latest } = data['dist-tags'];
        return latest;
    }
    catch (err) {
        throw new customErrors_1.CouldNotGetLatestVersion(err);
    }
}
exports.default = getLatestVersion;
