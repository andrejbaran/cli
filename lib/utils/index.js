"use strict";
/**
 * @author: JP Lew (jp@cto.ai)
 * @date: Thursday, 16th May 2019 10:22:25 am
 * @lastModifiedBy: JP Lew (jp@cto.ai)
 * @lastModifiedTime: Thursday, 16th May 2019 12:30:31 pm
 * @copyright (c) 2019 CTO.ai
 */
Object.defineProperty(exports, "__esModule", { value: true });
var asyncPipe_1 = require("./asyncPipe");
exports.asyncPipe = asyncPipe_1.asyncPipe;
exports._trace = asyncPipe_1._trace;
var onExit_1 = require("./onExit");
exports.onExit = onExit_1.onExit;
var getOpUrl_1 = require("./getOpUrl");
exports.getOpImageTag = getOpUrl_1.getOpImageTag;
exports.getOpUrl = getOpUrl_1.getOpUrl;
