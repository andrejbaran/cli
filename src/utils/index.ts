/**
 * @author: JP Lew (jp@cto.ai)
 * @date: Thursday, 16th May 2019 10:22:25 am
 * @lastModifiedBy: JP Lew (jp@cto.ai)
 * @lastModifiedTime: Thursday, 6th June 2019 6:27:27 pm
 * @copyright (c) 2019 CTO.ai
 */

export { asyncPipe, _trace } from './asyncPipe'
export { validateEmail, validChars } from './validate'
export { onExit } from './onExit'
export { getOpImageTag, getOpUrl } from './getOpUrl'
export { handleMandatory, handleUndefined } from './guards'
