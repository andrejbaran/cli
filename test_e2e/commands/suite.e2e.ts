/**
 * @author: JP Lew (jp@cto.ai)
 * @date: Friday, 24th May 2019 1:41:52 pm
 * @lastModifiedBy: JP Lew (jp@cto.ai)
 * @lastModifiedTime: Wednesday, 12th June 2019 4:58:08 pm
 * @copyright (c) 2019 CTO.ai
 */

import fs from 'fs-extra'
import { run, sleep, cleanup, signin, signup } from '../utils/cmd'
import {
  ENTER,
  SPACE,
  DOWN,
  NEW_OP_NAME,
  NEW_OP_DESCRIPTION,
  NEW_USER_EMAIL,
  NEW_USER_PASSWORD,
  NEW_USER_NAME,
  EXISTING_USER_EMAIL,
  EXISTING_USER_PASSWORD,
} from '../utils/constants'

// give the suite max 5 minutes to complete
jasmine.DEFAULT_TIMEOUT_INTERVAL = 1000 * 60 * 5

beforeEach(async () => {
  try {
    await run(['account:signout'])
  } catch (err) {
    console.error(err)
  }
})

afterAll(async () => {
  // avoid jest open handle error
  await sleep(500)
})

// test is redundant but useful for debugging
test.skip('it should signup', async () => {
  console.log('it should signup')
  const result = await signup(NEW_USER_EMAIL, NEW_USER_NAME, NEW_USER_PASSWORD)

  expect(result).toMatchSnapshot()
  await cleanup()
})

// test is redundant but useful for debugging
test.skip('it should signin', async () => {
  console.log('it should signin')
  const result = await signin(EXISTING_USER_EMAIL, EXISTING_USER_PASSWORD)

  expect(result).toMatchSnapshot()
  expect(result).toContain('Welcome back')
})

test('it should signup, signin, init, build, publish, search', async () => {
  console.log('it should signup, signin, init, build, publish, search')

  await signup(NEW_USER_EMAIL, NEW_USER_NAME, NEW_USER_PASSWORD)
  await sleep(500)

  await signin(NEW_USER_EMAIL, NEW_USER_PASSWORD)
  await sleep(500)

  console.log('ops init')
  try {
    const initRes = await run(
      ['init'],
      [SPACE, ENTER, NEW_OP_NAME, ENTER, NEW_OP_DESCRIPTION, ENTER],
    )
    expect(initRes.toLowerCase()).toContain('success')
    expect(initRes).toContain('ðŸš€ To test your op run: $ ops run t_my_new_op')
  } catch (e) {
    throw e
  }

  await sleep(500)

  console.log(`ops build ${NEW_OP_NAME}`)
  try {
    const buildRes = await run(['build', NEW_OP_NAME])
    expect(buildRes.toLowerCase()).toContain('successfully built')
  } catch (e) {
    throw e
  }

  await sleep(500)

  console.log(`ops publish ${NEW_OP_NAME}`)
  try {
    const publishRes = await run(['publish', NEW_OP_NAME])
    expect(publishRes.toLowerCase()).toContain('preparing:')
    expect(publishRes).toContain('has been published!')
  } catch (e) {
    throw e
  }
  await sleep(500)

  console.log('ops search')
  try {
    const searchRes = await run(['search'])
    await sleep(500)

    expect(searchRes).toContain(NEW_OP_NAME)
  } catch (e) {
    throw e
  }
  await sleep(500)

  const pathToOp = `./${NEW_OP_NAME}`

  if (fs.existsSync(pathToOp)) {
    fs.removeSync(pathToOp)
    console.log(pathToOp, ' directory deleted successfully.')
  }

  await cleanup()
})
