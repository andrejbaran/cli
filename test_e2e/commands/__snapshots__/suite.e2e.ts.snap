// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should signin 1`] = `
"{ DEBUG: 'ops:*', NODE_ENV: 'staging' }
{ envConfig:
   { OPS_REGISTRY_HOST: 'registry.stg-platform.hc.ai',
     OPS_API_HOST: 'http://www.stg-platform.hc.ai/' } }

💻 CTO.ai Ops - The CLI built for Teams 🚀

👋 Welcome to the Ops CLI beta! 

Please login to get started.

 Enter email:  [15D[15C[2K[G Enter email:  e[16D[16C[2K[G Enter email:  e2[17D[17C[2K[G Enter email:  e2e[18D[18C[2K[G Enter email:  e2e_[19D[19C[2K[G Enter email:  e2e_e[20D[20C[2K[G Enter email:  e2e_ex[21D[21C[2K[G Enter email:  e2e_exi[22D[22C[2K[G Enter email:  e2e_exis[23D[23C[2K[G Enter email:  e2e_exist[24D[24C[2K[G Enter email:  e2e_existi[25D[25C[2K[G Enter email:  e2e_existin[26D[26C[2K[G Enter email:  e2e_existing[27D[27C[2K[G Enter email:  e2e_existing_[28D[28C[2K[G Enter email:  e2e_existing_u[29D[29C[2K[G Enter email:  e2e_existing_us[30D[30C[2K[G Enter email:  e2e_existing_use[31D[31C[2K[G Enter email:  e2e_existing_user[32D[32C[2K[G Enter email:  e2e_existing_user@[33D[33C[2K[G Enter email:  e2e_existing_user@c[34D[34C[2K[G Enter email:  e2e_existing_user@ct[35D[35C[2K[G Enter email:  e2e_existing_user@cto[36D[36C[2K[G Enter email:  e2e_existing_user@cto.[37D[37C[2K[G Enter email:  e2e_existing_user@cto.a[38D[38C[2K[G Enter email:  e2e_existing_user@cto.ai[39D[39C[2K[G Enter email:  e2e_existing_user@cto.ai[39D[39C
 Enter password:  [18D[18C[2K[G Enter password:  *[19D[19C[2K[G Enter password:  **[20D[20C[2K[G Enter password:  ***[21D[21C[2K[G Enter password:  ****[22D[22C[2K[G Enter password:  *****[23D[23C[2K[G Enter password:  ******[24D[24C[2K[G Enter password:  *******[25D[25C[2K[G Enter password:  ********[26D[26C[2K[G Enter password:  ********[26D[26C


👋 Welcome back e2e_existing_user!

👉 Type ops search to find ops or ops init to create your own! 

"
`;

exports[`it should signup 1`] = `
"
💻 CTO.ai Ops - The CLI built for Teams 🚀

👋 Welcome to the Ops CLI beta! 

❔ Let us know if you have questions...
📬  You can always reach us by email (mailto:support@cto.ai) (support@cto.ai).

⚡️  Let's get you started...
 
📩 Please enter your email →  
Enter Email [12D[12C[2K[1A[2K[1A[2K[G 
📩 Please enter your email →  
Enter Email t[13D[13C[2K[1A[2K[1A[2K[G 
📩 Please enter your email →  
Enter Email t_[14D[14C[2K[1A[2K[1A[2K[G 
📩 Please enter your email →  
Enter Email t_e[15D[15C[2K[1A[2K[1A[2K[G 
📩 Please enter your email →  
Enter Email t_em[16D[16C[2K[1A[2K[1A[2K[G 
📩 Please enter your email →  
Enter Email t_ema[17D[17C[2K[1A[2K[1A[2K[G 
📩 Please enter your email →  
Enter Email t_emai[18D[18C[2K[1A[2K[1A[2K[G 
📩 Please enter your email →  
Enter Email t_email[19D[19C[2K[1A[2K[1A[2K[G 
📩 Please enter your email →  
Enter Email t_email_[20D[20C[2K[1A[2K[1A[2K[G 
📩 Please enter your email →  
Enter Email t_email_n[21D[21C[2K[1A[2K[1A[2K[G 
📩 Please enter your email →  
Enter Email t_email_ne[22D[22C[2K[1A[2K[1A[2K[G 
📩 Please enter your email →  
Enter Email t_email_new[23D[23C[2K[1A[2K[1A[2K[G 
📩 Please enter your email →  
Enter Email t_email_new_[24D[24C[2K[1A[2K[1A[2K[G 
📩 Please enter your email →  
Enter Email t_email_new_u[25D[25C[2K[1A[2K[1A[2K[G 
📩 Please enter your email →  
Enter Email t_email_new_us[26D[26C[2K[1A[2K[1A[2K[G 
📩 Please enter your email →  
Enter Email t_email_new_use[27D[27C[2K[1A[2K[1A[2K[G 
📩 Please enter your email →  
Enter Email t_email_new_user[28D[28C[2K[1A[2K[1A[2K[G 
📩 Please enter your email →  
Enter Email t_email_new_user@[29D[29C[2K[1A[2K[1A[2K[G 
📩 Please enter your email →  
Enter Email t_email_new_user@c[30D[30C[2K[1A[2K[1A[2K[G 
📩 Please enter your email →  
Enter Email t_email_new_user@ct[31D[31C[2K[1A[2K[1A[2K[G 
📩 Please enter your email →  
Enter Email t_email_new_user@cto[32D[32C[2K[1A[2K[1A[2K[G 
📩 Please enter your email →  
Enter Email t_email_new_user@cto.[33D[33C[2K[1A[2K[1A[2K[G 
📩 Please enter your email →  
Enter Email t_email_new_user@cto.a[34D[34C[2K[1A[2K[1A[2K[G 
📩 Please enter your email →  
Enter Email t_email_new_user@cto.ai[35D[35C[2K[1A[2K[1A[2K[G ✓ Email t_email_new_user@cto.ai added![39D[39C
 
🖖 Create a username to get started →  
Enter Username [15D[15C[2K[1A[2K[1A[2K[G 
🖖 Create a username to get started →  
Enter Username t[16D[16C[2K[1A[2K[1A[2K[G 
🖖 Create a username to get started →  
Enter Username t_[17D[17C[2K[1A[2K[1A[2K[G 
🖖 Create a username to get started →  
Enter Username t_u[18D[18C[2K[1A[2K[1A[2K[G 
🖖 Create a username to get started →  
Enter Username t_us[19D[19C[2K[1A[2K[1A[2K[G 
🖖 Create a username to get started →  
Enter Username t_use[20D[20C[2K[1A[2K[1A[2K[G 
🖖 Create a username to get started →  
Enter Username t_user[21D[21C[2K[1A[2K[1A[2K[G 
🖖 Create a username to get started →  
Enter Username t_user_[22D[22C[2K[1A[2K[1A[2K[G 
🖖 Create a username to get started →  
Enter Username t_user_n[23D[23C[2K[1A[2K[1A[2K[G 
🖖 Create a username to get started →  
Enter Username t_user_ne[24D[24C[2K[1A[2K[1A[2K[G 
🖖 Create a username to get started →  
Enter Username t_user_new[25D[25C[2K[1A[2K[1A[2K[G ✓ Username t_user_new created![31D[31C
 
🔑 Let's create a password next →  
Enter your password [20D[20C[2K[1A[2K[1A[2K[G 
🔑 Let's create a password next →  
Enter your password *[21D[21C[2K[1A[2K[1A[2K[G 
🔑 Let's create a password next →  
Enter your password **[22D[22C[2K[1A[2K[1A[2K[G 
🔑 Let's create a password next →  
Enter your password ***[23D[23C[2K[1A[2K[1A[2K[G 
🔑 Let's create a password next →  
Enter your password ****[24D[24C[2K[1A[2K[1A[2K[G 
🔑 Let's create a password next →  
Enter your password *****[25D[25C[2K[1A[2K[1A[2K[G 
🔑 Let's create a password next →  
Enter your password ******[26D[26C[2K[1A[2K[1A[2K[G 
🔑 Let's create a password next →  
Enter your password *******[27D[27C[2K[1A[2K[1A[2K[G 
🔑 Let's create a password next →  
Enter your password ********[28D[28C[2K[1A[2K[1A[2K[G ✓ Password added! ********[27D[27C
 
🔑 Confirm your password:  [26D[27C[2K[1A[2K[G 
🔑 Confirm your password:  *[27D[28C[2K[1A[2K[G 
🔑 Confirm your password:  **[28D[29C[2K[1A[2K[G 
🔑 Confirm your password:  ***[29D[30C[2K[1A[2K[G 
🔑 Confirm your password:  ****[30D[31C[2K[1A[2K[G 
🔑 Confirm your password:  *****[31D[32C[2K[1A[2K[G 
🔑 Confirm your password:  ******[32D[33C[2K[1A[2K[G 
🔑 Confirm your password:  *******[33D[34C[2K[1A[2K[G 
🔑 Confirm your password:  ********[34D[35C[2K[1A[2K[G ✓ Password confirmed! ********[31D[31C


✅  Your account is setup! You can now build, run and share ops!
🎉  We just sent you an email with tips on how to get started!


💻  Now try running ops search to find some ops to run, or ops init to begin creating your own. 


"
`;
